### PR TITLE
poc: adoptedStylesheet conversion boogaloo

### PR DESCRIPTION
**Pull request by louis-bompart**
This PR demonstrate the proposed way forward to reduce our CSS duplication.
As discussed previously, it relies heavily on constructed & adopted stylesheets (with shims borrowed from lit/lit for retrocompatibility)

## The build process:
To avoid rewriting heaps of CSS, I wrote a piece of code that goes through a list of pcss files and then explores its imports.
Each file is processed by postcss individually into a typescript file that exports an array of strings (we strip the import)
In this array is spread the import of the dependent and then its own CSS.

It's not perfect by any account:
- repetition in the CSS rule is to be expected.
- `@apply` of custom class wont work
- mixins are unlikely to work either

However, it maximizes the DRY for the bundle without too much change in the actual source code, I think.

## The import/apply process:

Essentially, we adoptStylesheet if we can or can't and add them as style tags in the shadow root.

### Adopt method
Because the generated CSS can be chunky, I made the constructed stylesheet instance reusable by having them in a Map string-to-ConstructedStyleSheet.
To ensure instantiated stylesheets are garbage-collected, if possible, they are stored in the map as WeakRef.

### Style tag method
We need to handle the nonce. It's usually accessible in the bindings, but this time, we need it before the bindings are available.
So, I store the nonce in a singleton and reuse it to create the stylesheet.
The rest is as lit does it.

### CODE REVIEW

File: packages/atomic/scripts/postcss-build.cjs
Comment (louis-bompart) [inline packages/atomic/scripts/postcss-build.cjs:1]: need to be cjs 'cause tailwindcss/nested is a CJS

File: packages/atomic/src/utils/adoptedStyleSheets-utils.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  'replace' in CSSStyleSheet.prototype;
+
+function getStyleSheet(styleSheet: string): CSSStyleSheet {
+  let constructedStyleSheet: CSSStyleSheet | undefined;
+  if (styleSheetMap.has(styleSheet)) {
+    constructedStyleSheet = styleSheetMap.get(styleSheet)?.deref();
+  }
+  if (!constructedStyleSheet) {
+    constructedStyleSheet = new CSSStyleSheet();
+    constructedStyleSheet.replaceSync!(styleSheet);
```
Comment (louis-bompart) [inline packages/atomic/src/utils/adoptedStyleSheets-utils.ts:16]: - [ ] check promise w/ `.replace` for optimization/parallelization

File: packages/atomic/src/components/commerce/facets/atomic-commerce-category-facet/atomic-commerce-category-facet.tsx [hunk 2]
```diff
@@ -41,6 +44,7 @@ import {FacetShowMoreLess} from '../../../common/facets/facet-show-more-less/fac
 import {FacetValuesGroup} from '../../../common/facets/facet-values-group/facet-values-group';
 import {initializePopover} from '../../../common/facets/popover/popover-type';
 import {CommerceBindings as Bindings} from '../../atomic-commerce-interface/atomic-commerce-interface';
+import css from './atomic-commerce-category-facet.css';
```
Comment (fpbrault) [inline packages/atomic/src/components/commerce/facets/atomic-commerce-category-facet/atomic-commerce-category-facet.tsx:47]: Why not allCss here?

Comment (louis-bompart) [reply packages/atomic/src/components/commerce/facets/atomic-commerce-category-facet/atomic-commerce-category-facet.tsx:47]: in a `import ns from './path/to/file'`, `ns` can be anything, it's a concern of the importer.
`ns` is just the variable in which `default` from the export will be stored.
I think cssAll for the consumer is noisy, so I elected to call it `css` instead

Comment (fpbrault) [reply packages/atomic/src/components/commerce/facets/atomic-commerce-category-facet/atomic-commerce-category-facet.tsx:47]: Oh right, forgot about that!

File: packages/atomic/src/components/commerce/facets/atomic-commerce-facets/atomic-commerce-facets.css.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+// This file is generated by postcss-build.cjs
```
Comment (fpbrault) [inline packages/atomic/src/components/commerce/facets/atomic-commerce-facets/atomic-commerce-facets.css.ts:1]: Is the idea to generate these once as we migrate to this new method and then use twind and modify the css.ts files by hand? 

Or will we have to run postcss-build.cjs (hopefully automatically) whenever we make change to the pcss files?

**Pull request closed without merging**